File: library/src/test/java/me/proxer/library/enums/ProjectStateTest.java
package me.proxer.library.enums;

import me.proxer.library.ProxerTest;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

class ProjectStateTest extends ProxerTest {

    @Test
    void testDefault() throws IOException {
        final ProjectState projectState = api.moshi().adapter(ProjectState.class).fromJson("\"1\"");

        assertThat(projectState).isSameAs(ProjectState.FINISHED);
    }

    @Test
    void testFallback() throws IOException {
        final ProjectState projectState = api.moshi().adapter(ProjectState.class).fromJson("\"xyz\"");

        assertThat(projectState).isSameAs(ProjectState.UNKNOWN);
    }
}

File: library/src/test/kotlin/me/proxer/library/enums/ProjectStateTest.kt
package me.proxer.library.enums

import com.serjltt.moshi.adapters.FallbackEnum
import com.squareup.moshi.Moshi
import org.amshove.kluent.shouldBe
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class ProjectStateTest {

    private val adapter = Moshi.Builder()
        .add(FallbackEnum.ADAPTER_FACTORY)
        .build()
        .adapter(ProjectState::class.java)

    @Test
    fun testDefault() {
        adapter.fromJson("\"1\"") shouldBe ProjectState.FINISHED
    }

    @Test
    fun testFallback() {
        adapter.fromJson("\"xyz\"") shouldBe ProjectState.UNKNOWN
    }
}
